src/util/run.cpp

```diff
@@ -87,10 +87,34 @@ int run(
   const std::string &std_error)
 {
   #ifdef _WIN32
-  // we don't support stdin/stdout/stderr redirection on Windows
-  PRECONDITION(std_input.empty());
-  PRECONDITION(std_output.empty());
-  PRECONDITION(std_error.empty());
+  // we use the shell to do stdin/stdout/stderr redirection on Windows
+  if(!std_input.empty() || !std_output.empty() || !std_error.empty())
+  {
+    std::vector<std::string> new_argv = argv;
+    new_argv.insert(new_argv.begin(), "cl.exe");
+    new_argv.insert(new_argv.begin() + 1, "/c");
```

Comment:
I don't think `run` was meant to be limited to `cl.exe`?! It should equally be usable, e.g., with an SMT solver?!

Reply:
The SMT solvers are currently called with system(), but it would actually be a good idea to change this (will do separate PR).
Fixed!

Reply:
> Fixed!

It seems line 116 still has `cl.exe` hard-coded.